applications/graphs/views.py

```diff
@@ -494,6 +495,19 @@ def _update_graph(request, graph_id, graph={}):
 	authorization.validate(request, permission='GRAPH_UPDATE', graph_id=graph_id)
 	user_role = authorization.user_role(request)
 
+	if 'update_legend_format' in graph:
```

Comment:
Views should only do basic computations - passing request to the correct methods in controllers.py and returning a HTTPResponse. 
I believe a custom method like `graphs.update_graph_with_legend()` which does all the computation and returns a Graph dataset will make more sense.

Then this view could be reduced to -
```
if 'update_legend_format' in graph:
        return utils.serializer(graphs.update_graph_with_legend(request, graph_id=graph_id))
```